lib/archethic_web/graphql_schema/resolver.ex

```diff
@@ -33,6 +34,54 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
     end
   end
 
+  def get_token(address) do
+    genesis_address =
+      case TransactionChain.fetch_genesis_address_remotely(address) do
+        {:ok, genesis_address} -> genesis_address
+        _ -> nil
+      end
+
+    transaction_content =
+      case Archethic.search_transaction(address) do
+        {:ok, %Transaction{data: %TransactionData{content: content}, type: type}} ->
```

Comment:
Instead of doing if `type == :token`, you can pattern match the type of the transaction.
This will reduce the complexity of the code and make it easier to read/understand

Reply:
Yup that solves it